tools/darknet/darknet2ncnn.cpp

```diff
@@ -828,13 +828,14 @@ void load_weights(const char* filename, std::deque<Section*>& dnet)
             }
 
             if (s->layer_type == "Convolution")
-                read_to(s->weights, s->c * s->filters * s->size * s->size, fp);
+                read_to(s->weights, (size_t)(s->c) * s->filters * s->size * s->size, fp);
             else if (s->layer_type == "ConvolutionDepthWise")
                 read_to(s->weights, s->c * s->filters * s->size * s->size / s->groups, fp);
         }
     }
 
-    fclose(fp);
+    if (fp != NULL)
```

Comment:
I think this is an extra check. We have condition with NULL higher:

    FILE* fp = fopen(filename, "rb");
    if (fp == NULL)
        file_error(filename); // exit

Reply:
@proydakov Agree. I did this modification since Visual Studio 2019 give "warning hint" if not check. Trying to find reasonable explanation for this hint.